Source/HTTP/Curl/CurlProvider.cpp

```diff
@@ -124,7 +124,7 @@ HRESULT CALLBACK CurlProvider::CleanupAsyncProvider(XAsyncOp op, const XAsyncPro
 
         {
             std::lock_guard<std::mutex> lock{ provider->m_mutex };
-            provider->m_pendingMultiCleanups = provider->m_curlMultis.size();
+            provider->m_cleanupTasksRemaining = 1 + provider->m_curlMultis.size();
```

Comment:
nit: it might be worth a comment here describing what the +1 is for

Reply:
will do